initial orangefs work for sdp

AJMKing opened the pull request:
Branch for the initial orangefs work for SDP. Contains two new roles to setup and configure Orangefs, with the rpms, and client unit file.

Commit:
initial orangefs work for sdp

Commit:
initial orangefs work for sdp

oneswig commented on inventory.pvfs:
```diff
@@ -0,0 +1,21 @@
+# Ansible Shade uses OpenStack clients running locally
```
Inventory files should be auto-generated by the Heat role (which you didn't run, so that's fine).

I'm wondering where your DNS is coming from, have you tinkered with that?

What's interesting here is that we have overlapping roles: a node can be both server and client.  This needs a little bit of thinking (but probably, in the first instance, manual adjustment)

AJMKing replied:
i had SSH config point to the nodes and sort out the keys. So ansible will be resolving names that way. not sure how this works with heat.

oneswig commented on orange_setup/files/orangefs_client.service:
```diff
@@ -0,0 +1,10 @@
+[Unit]
```
As discussed, if this could get tucked into the RPM that'd be grand.

oneswig commented on orange_setup/tasks/main.yml:
```diff
@@ -0,0 +1,29 @@
+---
+- name: copy rpms to server
```
I don't think we should have the RPMs as artefacts in this repo, but how we handle them otherwise is TBD.  Best option may be to make a locally-accessible repo server which contains the OrangeFS RPMs, plus a few others we've got kicking around.

oneswig commented on orange_setup/tasks/main.yml:
```diff
@@ -0,0 +1,29 @@
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - /tmp/orangefs-2.9.7-1.fc25.x86_64.rpm
```
These two items appear to be the same...

AJMKing replied:
you meen the two RPMs? one should provide the server only one provides the client.

oneswig commented on orangefs/tasks/main.yml:
```diff
@@ -0,0 +1,87 @@
+  when: inventory_hostname in groups['orange-data']
+
+- name: setup mdraid 5 on nvmes
+  command: mdadm --create --verbose /dev/md0 --level=5 --raid-devices=4 /dev/nvme0n1 /dev/nvme1n1 /dev/nvme2n1 /dev/nvme3n1 creates=/dev/md0
```
If /dev/md0 already exists, does this fail?  If it doesn't, it'll get flattened in the next task.  If it does, it means the playbook won't rerun.

AJMKing replied:
if md0 exists  the command wont run. if it dose not, then it will try to build the array. If the case that md0 was deleted or the mdadm stop, but raid superblocks exist, then the playbook will fail. Should probably brakeout this into a role that sets up mdraid, since lvm is also an option.

oneswig commented on orangefs/tasks/main.yml:
```diff
@@ -0,0 +1,87 @@
+---
```
It might be tidier syntax to conditionally include task files for client, server, etc. after the common tasks here.

oneswig commented on orangefs/tasks/main.yml~:
```diff
@@ -0,0 +1,90 @@
+---
```
You Emacs users have no shame :-)

oneswig commented on orange_setup/tasks/main.yml~:
```diff
@@ -0,0 +1,20 @@
+---
```
Temporary Emacs file?

oneswig reviewed (changes requested):
Thanks Ali, some good stuff here but we'll need to reshape a bit here and there.  I'll try to get the Heat template work done and then I'll try to do a test run of the code checked in.

Commit:
moved orange into roles and removed tildas

Commit:
moved orange into roles and removed tildas

oneswig committed:
Rebased Alasdairs OrangeFS work onto Heat-managed infrastructure and
using the new directory structure for p3-appliances.

Generalised some components (a bit) to enable cluster deployment with
FC25 images.

AJMKing committed:
Merge pull request #1 from oneswig/orangefs

Orangefs

Commit:
added in mrlesmithjr/ansible-mdadm to replase static mdraid setup on orangefs

Commit:
add files for mdadm

Commit:
add tests for mdraid provided by module

Commit:
removed mdraid work to be replaced with lvm as this is provided with ansible

Commit:
updated merge

oneswig committed:
Move Alasdair's latest work into new directory layout

AJMKing committed:
Merge pull request #2 from oneswig/orangefs

Move Alasdair's latest work into new directory layout

oneswig committed:
More tweaks, almost there

oneswig committed:
Generate a consistent PVFS2 config across the OrangeFS cluster

AJMKing committed:
Merge pull request #3 from oneswig/orangefs

More tweaks, almost there - getting unknown protocol (-92) on mount

oneswig committed:
Final fixes, deliver a working filesystem from baseline infrastrucutre

AJMKing committed:
Merge pull request #4 from oneswig/orangefs

Final fixes, deliver a working filesystem from baseline infrastructure
